wgpu-types/src/lib.rs

```diff
@@ -6057,20 +6071,24 @@ fn test_max_mips() {
 /// Corresponds to [WebGPU `GPUTextureDescriptor`](
 /// https://gpuweb.github.io/gpuweb/#dictdef-gputexturedescriptor).
 #[repr(C)]
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[derive(bon::Builder, Clone, Debug, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct TextureDescriptor<L, V> {
+pub struct TextureDescriptor<L: Default, V: Default> {
     /// Debug label of the texture. This will show up in graphics debuggers for easy identification.
+    #[builder(default, into)]
```

Comment:
Unfortunately, it's not obvious, but optional fields, that use a generic type are discouraged. This is because it leads to weakened type inference.

In short, if you minimize the problem it boils down to this analogous pattern:

```rust
fn example<T>(_value: Option<T>) {}

fn main() {
    example(None);
}
```

This code doesn't compile with the error:
``` example(None);
  |     ^^^^^^^ ---- type must be known at this point
  |     |
  |     cannot infer type of the type parameter `T` declared on the function `example`
```

This same problem would occur if you use `TextureDescriptor::builder()` and omit to cal the `label` setter. The compiler will not be able to infer the generic type `L` in such case.

I described this problem in detail [here](https://bon-rs.com/guide/patterns/optional-generic-members), feel free to check that explanation on your leisure.